### DISCUSSION TITLE
Better client representation for Postgres RANGE types

### DISCUSSION THREAD

**Discussion by isaacharrisholt** [supabase/supabase#17398] [Feature Requests]
Hi there,

We recently started working with Postgres `range` types in our Supabase DB. Sadly, PostgREST, and therefore Supabase, doesn't have great support for these types. Selecting a range will return the string representation of the range, e.g. a date range: `[2023-01-01,2023-04-05)`.

This is technically an accurate representation, but it would be nice if the client libraries could wrap these somehow so they're a bit more user-friendly. There's [a JS library for this](https://github.com/martianboy/postgres-range) with a nice API that could be emulated somewhat.

Let me know your thoughts :)

**Comment by steve-chavez** [comment]
Hello,

There's a couple of ways you can modify the range type format for all requests:

- https://postgrest.org/en/stable/how-tos/working-with-postgresql-data-types.html#casting-a-range-to-a-json-object, needs casting `select(your_range::json)` everytime.
- https://postgrest.org/en/stable/references/api/domain_representations.html, works transparently without casting, but requires a wrapper DOMAIN type on your column (`CREATE DOMAIN myrange as range`). 

Does that help?

**Reply by isaacharrisholt** [reply]
Yes! I saw these when I was looking at solutions myself. I just figured it would be nice to have something like this baked into Supabase. 

These also only work one way, so you still need to serialise to a range string representation on write.

**Reply by steve-chavez** [reply]
> These also only work one way, so you still need to serialise to a range string representation on write.

Actually with "domain representations" you can have the same format work for writes.

https://postgrest.org/en/stable/references/api/domain_representations.html#domain-request-body-format

**Reply by isaacharrisholt** [reply]
Ooh interesting, thanks! Do you know how this would interact with Supabase's generated types? ATM the TypeScript ones just use `string` for ranges.
